### ISSUE TITLE
Is it possible to cut cell content based on cell width

### ISSUE THREAD

**Issue by bmihelac** [bpampuch/pdfmake#264]
Currently pdfmake  wrap content if it does not fit in given cell and if text cannot be wrapped it is diplay and overflows outside of bounds. 

Is it possible to hide overflow content?

Example:

    var dd = {
        content: [
                    {
                            table: {
                                widths: [50],
                                    body: [
                                            [{text: 'ColumnColumnColumnColumnColumnColumnColumn'}]
                                    ]
                            }
                    },
        ],

    }

**Comment by Wabbala** [comment]
I do not think that hyphenation is supported in pdfmake. It would wrap if
your text is Column Column Column Column Column Column Column.

**Comment by bmihelac** [comment]
@Wabbala I do not want to hyphenate word, I want to cut excessive text that does not fit specified width of cell.

**Comment by jthoenes** [comment]
That's currently not supported in pdfmake.

Happy to accept a pull request. Let me know when you need help for it.
 Am 17.04.2015 14:49 schrieb "Bojan Mihelac" <notifications@github.com>:

**Comment by bmihelac** [comment]
Ok thanks for answer, I will try to make PR for this.

Looking at [pdfkit documentation](http://pdfkit.org/docs/text.html#the_basics) under **Text styling** there are two options that should be of interest for implementing this option:

* width - the width that text should be wrapped to (by default, the page width minus the left and right margin)
* height - the maximum height that text should be clipped to

I guess we should add width and height options to styling. Am I on the right path?

**Comment by jthoenes** [comment]
No, we're not using the pdfkit ``text`` method for writing text. See https://github.com/bpampuch/pdfmake/blob/master/src/printer.js#L257.

Have a look at https://github.com/bpampuch/pdfmake/blob/master/src/layoutBuilder.js and https://github.com/bpampuch/pdfmake/blob/master/src/tableProcessor.js.

Detect, when a line will be to long (you can before you are in the printer) and react to it.

**Comment by bmihelac** [comment]
Pull request allows setting of ``maxHeight``. It works in with text object so it can be used in columns and tables as well.

This is not exactly what I was asking for but still can be usable.

Example input:

    var dd = {
      content: [
      {
        text: 'Max one line - Another paragraph, this time a little bit longer to make sure, this line will be divided into at least two lines',
        maxHeight: 15,
        marginBottom: 15
        },
      {
        text: 'Max two lines - Another paragraph, this time a little bit longer to make sure, this line will be divided into at least two lines Another paragraph, this time a little bit longer to make sure, this line will be divided into at least two lines',
        maxHeight: 30,
        }
      ]
    }

Result:

![screen shot 2015-06-03 at 14 07 31](https://cloud.githubusercontent.com/assets/13813/7959491/dc138316-09f9-11e5-993e-da2171501f72.png)

**Comment by zamb3zi** [comment]
Here's [my solution](https://github.com/bpampuch/pdfmake/issues/1733#issuecomment-1009736948) at #1733 for this that doesn't use the `maxHeight` property.
It measures the text width and truncates with ellipses as necessary.
